conjure/ui/widgets/option_widget.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2016 Canonical, Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from enum import Enum
+from urwid import (CheckBox, Columns, connect_signal, Divider, GridFlow,
+                   IntEdit, Pile, Text, WidgetWrap)
+from ubuntui.widgets.buttons import PlainButton
+from ubuntui.widgets.input import StringEditor
+from ubuntui.widgets.hr import HR
+from ubuntui.utils import Color, Padding
+import logging
+
+log = logging.getLogger('conjure')
+
+
+class OptionType(Enum):
+    BOOLEAN = 0
+    STRING = 1
+    INT = 2
+
+
+def strip_solo_dots(s):
+    ls = s.split("\n")
+    rl = []
+    for l in ls:
+        if l == ".":
+            rl.append("")
+        else:
+            rl.append(l)
+    return "\n".join(rl)
+
+
+class OptionWidget(WidgetWrap):
+
+    def __init__(self, name, optype, description, default,
+                 current_value=None, value_changed_callback=None):
+        self.name = name
+        self.optype = OptionType.__members__[optype.upper()]
+        self.description = description
+        self.default = default
+        self.current_value = current_value or default
+        w = self.build_widgets()
+        self.value_changed_callback = value_changed_callback
+        super().__init__(w)
+        self.update()
+
+    def selectable(self):
+        return True
+
+    def build_widgets(self):
+        desc_text = Text(["\n", strip_solo_dots(self.description)])
+
+        self.reset_button = PlainButton("Reset to Default", self.do_reset)
+        if self.optype == OptionType.BOOLEAN:
+            self.control = CheckBox('', state=bool(self.current_value))
+            self.wrapped_control = self.control
```

Comment:
Note - using the radio_input style here looked weird. it had a dark gray background for like 60 chars when the actual check box is only 3.